notebooks/COS/Extract/Extract.ipynb

```diff
@@ -0,0 +1,1580 @@
+    "\n",
+    "*Note* that we can ignore the UnitsWarning."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "minus-continent",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set up figure\n",
+    "fig = plt.figure(figsize = (20, 7))\n",
+    "gs = fig.add_gridspec(nrows = 5, ncols = 3) # Using gridspec to let us control panel sizes and locations\n",
+    "\n",
+    "for i in range(3):\n",
+    "    ax0 = fig.add_subplot(gs[0:4,i])\n",
+    "    ax1 = fig.add_subplot(gs[4:5,i])\n",
+    "    new_wvln, new_flux = Table.read('./output/calcos_run1/le4b04010_x1dsum.fits')[i]['WAVELENGTH', 'FLUX']\n",
+    "    old_wvln, old_flux, seg = Table.read(old_x1dsum)[i]['WAVELENGTH', 'FLUX', 'SEGMENT']\n",
+    "    \n",
+    "    # Interpolate the new wvln onto the old wvln's sampling:\n",
+    "    new_flux_interp = interp1d(x = new_wvln, y = new_flux, fill_value=\"extrapolate\")(old_wvln)\n",
+    "\n",
+    "    # give max difference:\n",
+    "    print(f\"Stripe {seg} differs by up to: \\\n",
+    "    {max(new_flux - old_flux)/np.mean(abs(old_flux)):.3f}%\")\n",
+    "\n",
+    "    # Plotting - upper panel\n",
+    "    ax0.plot(new_wvln,new_flux, linewidth = .5, label = '$New$ extracted spectrum')\n",
+    "    ax0.plot(old_wvln,old_flux, linewidth = .5, label = '$Original$ extracted spectrum')\n",
+    "    # Plotting - lower panel\n",
+    "    ax1.plot(new_wvln,old_flux - new_flux_interp, linewidth = .5, label = 'Residual')\n",
+    "    # Some formatting:\n",
+    "    ax0.set_title(f\"Segment {seg}\", fontsize = 20)\n",
+    "    ax0.set_xticks([])\n",
+    "    ax0.legend(loc = 'lower center')\n",
+    "    ax1.legend(loc = 'lower center')\n",
+    "    if i == 0: # Add axis labels to the plot\n",
+    "        ax0.set_ylabel(\"Flux\\n[$erg\\ \\AA^{-1}\\ cm^{-2}\\ s^{-1}$]\", fontsize = 20)\n",
+    "    if i == 1:\n",
+    "        plt.xlabel(\"Wavelength\", fontsize = 20)\n",
+    "plt.suptitle(\"Fig 3.1\\nComparing the old and new extracted spectra for each segment\", fontsize = 25)\n",
+    "plt.tight_layout()\n",
+    "plt.savefig(plotsdir+\"comp_extracted.png\", dpi = 200)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "brown-commodity",
+   "metadata": {},
+   "source": [
+    "<a id = fuvE></a>\n",
+    "# 4. Example using FUV data\n",
+    "\n",
+    "Let's go through doing all of the above with an FUV dataset and corresponding FUV XTRACTAB."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "valid-exchange",
+   "metadata": {},
+   "source": [
+    "First download the FUV data, we'll grab an FUV/G160M/C1533 dataset from the same proposal as the NUV dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "acquired-rehabilitation",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pl = Obs.get_product_list(Obs.query_criteria(proposal_id=15869, obs_id = 'LE4B01040'))\n",
```

Comment:
```suggestion
    "pl = Obs.get_product_list(Obs.query_criteria(proposal_id=15869, obs_id='LE4B01040'))\n",
```
(this is a good example of the inconsistent application of the space-around-equals guideline I mentioned earlier)